waspc/data/Generator/templates/server/src/auth/providers/email/resetPassword.ts

```diff
@@ -1,29 +1,50 @@
 import { Request, Response } from 'express';
-import { findUserBy, verifyToken } from "../../utils.js";
-import { updateUserPassword } from "./utils.js";
+import {
+    createProviderId,
+    findAuthIdentity,
+    updateAuthIdentityProviderData,
+    verifyToken,
+    deserializeAndSanitizeProviderData,
+} from "../../utils.js";
 import { ensureTokenIsPresent, ensurePasswordIsPresent, ensureValidPassword } from "../../validation.js";
 import { tokenVerificationErrors } from "./types.js";
+import HttpError from '../../../core/HttpError.js';
 
 export async function resetPassword(
     req: Request<{ token: string; password: string; }>,
     res: Response,
-): Promise<Response<{ success: true } | { success: false; message: string }>> {
-    const args = req.body || {};
+): Promise<Response<{ success: true }> | void> {
+    const args = req.body ?? {};
     ensureValidArgs(args);
 
     const { token, password } = args;
     try {
-        const { id: userId } = await verifyToken(token);
-        const user = await findUserBy({ id: userId });
-        if (!user) {
-            return res.status(400).json({ success: false, message: 'Invalid token' });
+        const { email, token: resetToken } = await verifyToken<{ email: string, token: string }>(token);
+
+        const providerId = createProviderId('email', email);
+        const authIdentity = await findAuthIdentity(providerId);
+        if (!authIdentity) {
+            throw new HttpError(400, "Invalid token");
+        }
+        
+        const providerData = deserializeAndSanitizeProviderData<'email'>(authIdentity.providerData);
+
+        if (resetToken !== providerData.passwordResetToken) {
```

Comment:
I wanted to beef up the security with keeping the token that needs to be provided. But ... I realised through some of my research that makes only sense if you use the token to find the user in the database.

Since our token in stored in a quasi-JSON field i.e. a `String` field, we can't query by the token which renders it useless to have in the DB. I've reverted the change and I'm no longer storing the token in the database.